numpy/lib/npyio.py

```diff
@@ -86,10 +86,12 @@ def __dir__(self):
         return object.__getattribute__(self, '_obj').keys()
 
 
-def zipfile_factory(*args, **kwargs):
+def zipfile_factory(file, mode='r', *args, **kwargs):
```

Comment:
OK, that makes sense. I'll work on it - probably by changing it back, because I don't think this is important enough to warrant a deprecation cycle.